x-parser.js

```diff
@@ -228,99 +228,112 @@ export class XParser {
-    ['#125', 'Malformed attribute text — attribute names must be alphanumeric, must be lowercase, must not start or end with hyphens, and cannot start with a number — and, attribute values must be enclosed in double-quotes.'],
-    ['#126', 'Malformed boolean attribute interpolation — attribute names must be alphanumeric, must be lowercase, must not start or end with hyphens, and cannot start with a number — and, attribute values must be enclosed in double-quotes.'],
-    ['#127', 'Malformed defined attribute interpolation — attribute names must be alphanumeric, must be lowercase, must not start or end with hyphens, and cannot start with a number — and, attribute values must be enclosed in double-quotes.'],
-    ['#128', 'Malformed attribute interpolation — attribute names must be alphanumeric, must be lowercase, must not start or end with hyphens, and cannot start with a number — and, attribute values must be enclosed in double-quotes.'],
-    ['#129', 'Malformed property interpolation — property names must be alphanumeric, must be lowercase, must not start or end with underscores, and cannot start with a number — and, property values must be enclosed in double-quotes.'],
-    ['#130', 'Malformed closing quote to a bound attribute or property. Enclosing quotes must be simple, double-quotes.'],
-
-    ['#140', 'CDATA sections are forbidden. Use html entities (character encodings) instead.'],
-
-    ['#150', 'Improper javascript escape (\\\\, \\$, \\`, \\x, \\u, \\t, \\n, etc.) in raw template string. Only valid HTML entities (character references) are supported in html as code points. Use “&bsol;” for “\\”, “&dollar;” for “\\$”, and “&grave;” for “\\`”. Use literal characters (e.g., newlines) to enter newlines in your templates.'],
-    ['#151', 'Malformed hexadecimal character reference (html entity) or ambiguous ampersand.'],
-    ['#152', 'Malformed html comment. Comments cannot start with “>” or “->” characters, they cannot include a set of “--” characters, and they cannot end with a “-” character.'],
-    ['#153', 'Forbidden html element used — this parser is opinionated about which elements are allowed in order to reduce complexity and improve performance.'],
-    ['#154', 'Unmatched closing tag at the end of your template. To avoid unintended markup, non-void tags must explicitly be closed.'],
-    ['#155', 'Mismatched closing tag used. To avoid unintended markup, non-void tags must explicitly be closed and all closing tag names must be a case-sensitive match.'],
-    ['#156', 'Forbidden, nontrivial interpolation of <textarea> tag used. Only basic interpolation is allowed — e.g., <textarea>${…}</textarea>.'],
-    ['#157', 'Forbidden declarative shadow root used (e.g., `<template shadowrootmode="open">`).'],
-  ]);
+  static #getErrorMessage(key) {
+    switch (key) {
```

Comment:
If you `switch` on this stuff, this will all get enforced by our code coverage targets. I had initially started trying to duplicate this mapping and enforce it in our tests… but that’s brittle. It’s really nice that this will just _always_ be enforced moving forward.